src/middleware/loadUser.ts

```diff
@@ -0,0 +1,55 @@
+import { Request, Response, NextFunction } from 'express';
+import config from '../config/app';
+import User from '../models/users';
+
+interface IAuthResponse {
```

Comment:
Should IAuthResponse extend the IUser interface?   I think it should but am willing to listen to arguments to the contrary.

Reply:
I'm not sure, but I think that the IAuthResponse holds elements that are not part of the IUser interface as well as the IUser interface has elements that IAuthResponse doesnt. Even though I am a big fan of inheritance, it doesnt seem like IUser interface is a good candidate to be a base class for IAuthResponse.

Reply:
Shouldn't authenticated users have a one-to-one association with the users in our database?   The users in our database use the IUser interface?    Do you know of a better way to make the association?

Reply:
I may be thinking about this backwards.   Go ahead and merge your changes if you feel confident that they won't break anything.